### PR TITLE
Feature/2871 16 remove alternating rows of table

### LINKED ISSUE #16: Remove table alternating row color
#### Describe the desired behavior

Make the alternating row color optional. Different projects have different needs.

#### Describe the current behavior

It alternates.

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes #16 

#### Changes proposed in this Pull Request:
- Removed alternate table row colors styles

#### Screenshots:
Screenshot 2022-02-01 at 4 43 44 PM

#### To Test:
- yarn start
- Open Material Components --> Data Display --> Table

### CODE REVIEW

**Review by jeffvg — APPROVED** [review]

File: CommonOverrides.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## Zebra Striping Table
```
Comment (huayunh) [inline CommonOverrides.md:1]: I like this doc! I think it is a great idea.

Comment (huayunh) [reply CommonOverrides.md:1]: Can consider adding an image to explain the difference too.

File: CommonOverrides.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+Table styles can be overridden to have alternate colors for table rows.
+
+```
+"&:nth-of-type(odd):not($selected)": {
+    backgroundColor: BLUIColors.darkBlack[300],
+    "&$hover:hover": {
+        backgroundColor: Color(BLUIColors.darkBlack[300])
+            .mix(Color(MediumBlackBackground), 0.5)
```
Comment (huayunh) [inline CommonOverrides.md:10]:
```suggestion
```ts
import Color from 'color';
import * as BLUIColors from '@brightlayer-ui/colors';
...

"&:nth-of-type(odd):not($selected)": {
    backgroundColor: BLUIColors.darkBlack[300],
    "&$hover:hover": {
        backgroundColor: Color(BLUIColors.darkBlack[300])
            .mix(Color(MediumBlackBackground), 0.5)
```

File: src/blueTheme.ts [hunk 3]
```diff
@@ -667,20 +667,12 @@ export const blueTheme: ThemeOptions = {
         },
         MuiTableRow: {
             root: {
-                backgroundColor: BLUIColors.white[100],
+                backgroundColor: ThemeColors.background.paper,
```
Comment (huayunh) [inline src/blueTheme.ts:670]: I would just drop this line here

File: src/blueDarkTheme.ts [hunk 4]
```diff
@@ -725,20 +725,12 @@ export const blueDarkTheme: ThemeOptions = {
         MuiTableRow: {
             root: {
                 color: ThemeColors.text.primary,
-                backgroundColor: BLUIColors.darkBlack[300],
+                backgroundColor: ThemeColors.background.paper,
```
Comment (huayunh) [inline src/blueDarkTheme.ts:728]: maybe drop this too.

**Review by huayunh — APPROVED** [review]

File: CommonOverrides.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import * as BLUIColors from '@brightlayer-ui/colors';
+...
+"&:nth-of-type(odd):not($selected)": {
+    backgroundColor: BLUIColors.darkBlack[300],
+    "&$hover:hover": {
+        backgroundColor: Color(BLUIColors.darkBlack[300])
+            .mix(Color(MediumBlackBackground), 0.5)
+            .string(),
+    },
+}
```
Comment (joebochill) [inline CommonOverrides.md:16]: You are referring to a bunch of variables in here that are not defined...

**Review by joebochill — CHANGES_REQUESTED** [review]

File: CommonOverrides.md (hunk 5 above)
Comment (joebochill) [reply CommonOverrides.md:16]: This needs to be an actual working example that someone could drop into their project (or it should be an example of how to put this override into the theme, in the createTheme function, if they want to use it everywhere throughout their app)

File: CommonOverrides.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
+## Zebra Striping Table
+
+Table styles can be overridden to have alternate colors for table rows.
+
+<div style="width: 100%; text-align: center">
+    <img width="100%" style="max-width: 600px" alt="Zebra Striping Table" src="./images/zebraStripingTable.png">
```
Comment (joebochill) [inline CommonOverrides.md:6]: Can we change this image to use the actual recommended colors? This definitely looks like zebra stripes, but I don't think we want to be encouraging anyone to actually use colors like this.

File: CommonOverrides.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    tableRow: {
+        backgroundColor: BLUIColors.white[100],
+        "&:hover": {
+            backgroundColor: Color(BLUIColors.white[100]).mix(Color(BLUIColors.black[50]), 0.5).string()
+            },
+        "&:nth-of-type(odd)": {
+            backgroundColor: BLUIColors.black[900],
+        "&:hover": {
+            backgroundColor: Color(BLUIColors.white[100]).mix(Color(BLUIColors.black[50]), 0.5).string()
+            }
```
Comment (joebochill) [inline CommonOverrides.md:29]: These are not the right colors...did you mix up light and dark theme values?

image

File: CommonOverrides.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+                                    .mix(Color(BLUIColors.black[50]), 0.5)
+                                    .string(),
+                            },
+                            '&:nth-of-type(odd)': {
+                                backgroundColor: BLUIColors.black[900],
+                                '&:hover': {
+                                    backgroundColor: Color(BLUIColors.black[900])
+                                        .mix(Color(BLUIColors.black[50]), 0.5)
+                                        .string(),
+                                },
```
Comment (joebochill) [inline CommonOverrides.md:73]: Same thing here with the colors.

**Review by joebochill — COMMENTED** [review]
Example updates look good, we just need to make sure the colors are correct.

**Comment by jeffvg** [comment]
looks good both light and dark theme

**Review by joebochill — APPROVED** [review]

**Pull request merged**